lib/GenomeFileUtil/core/GenomeUtils.py

```
@@ -229,3 +229,152 @@ def check_full_contig_length_or_multi_strand_feature(feature, is_transpliced, co
     if len(strand_set) > 1 and not is_transpliced:
         feature["warnings"] = feature.get('warnings', []) + [warnings["both_strand_coordinates"]]
     return feature 
+
+def check_feature_ids_uniqueness(genome):
+    """
+    Tests that all feature ids in a genome are unique across all 4 feature type lists
+    Returns dict of Non Unique IDS and the counts associated with them. 
+    If all IDS are unique, it then returns an empty dict
+    """
+    unique_feature_ids = set()
+    duplicate_feature_id_counts = dict()
+    feature_lists = ["features","cdss","mrnas","non_coding_features"]
+
+    for feature_list in feature_lists:
+        for feature in genome[feature_list]:
+            if feature["id"] in unique_feature_ids:
+                if feature["id"] in duplicate_feature_id_counts:
+                    duplicate_feature_id_counts[feature["id"]] += 1
+                else:
+                    duplicate_feature_id_counts[feature["id"]] = 2
+            else:
+                unique_feature_ids.add(feature["id"])
+    if len(unique_feature_ids) == 0:
+        duplicate_feature_id_counts["WARNING NO FEATURE IDS FOUND IN GENOME."] = 0
```

This should raise an exception not return a dict